### PR TITLE
Clarify that we no longer allow correlated properties to have a null …

### PR DESCRIPTION
**Pull request by andreasohlund**
…or default value

And how to workaround it using a custom finder

### CODE REVIEW

File: nservicebus/upgrades/5to6/handlers-and-sagas.md [hunk 1]
```diff
@@ -107,6 +107,8 @@ snippet:5to6-ConfigureHowToFindSaga
 
 Version 6 automatically correlates incoming message properties to its saga data counterparts. Any saga data correlation in the message handler code can be safely removed. Correlated properties (for existing saga instances) will not be changed once set.
 
+NOTE: NServiceBus will no longer allow a correlated property to have a null/default value. Use a custom finder for the relevant message to override this validation.
```
Comment (SimonCropp) [inline nservicebus/upgrades/5to6/handlers-and-sagas.md]: "No longer" is ambiguous.

Comment (SimonCropp) [inline nservicebus/upgrades/5to6/handlers-and-sagas.md]: Perhaps link here https://msdn.microsoft.com/en-us/library/83fhsxwc.aspx

Comment (SimonCropp) [inline nservicebus/upgrades/5to6/handlers-and-sagas.md]: Add the exception message

**Comment by ramonsmits** [comment]
Just thinking about this... Do we raise an exception if we receive a message that has a value in the correlation property that is a default value? Its a bit strange that we would not allow to persist it but still query using the default value and execute the handler.

**Comment by SimonCropp** [comment]
@Particular/nservicebus-maintainers can one of you chime in on @ramonsmits comment above

**Comment by andreasohlund** [comment]
We don't check the message so it would blow up "with default value not
allowed" when persisting the saga.

But again I don't see the use case for correlating on a null/default value?

The same workaround still applies: Create a custom finder

**Comment by andreasohlund** [comment]
And we can always put out a minor release that would provide a "opt-out"
setting should users complain. This way we will be getting the feedback at
least? (perhaps there is a valid use case)

That said there might be one for "singleton" saga where they might have
used a Guid.Default or something to always hit the same saga but lets see?

**Comment by SimonCropp** [comment]
@andreasohlund but if we get a message and the saga config says "use this prop on message" and that message prop is default shouldnt that be an exception state?

**Comment by andreasohlund** [comment]
What if the user override? (they can still change the corr id if they want)

```
 
if(message.OrderId == null)
{
    this.Data.OrderId = someOtherId;
}

```

The current code will not throw in the above scenario. I do agree that this is a funky scenario?

**Comment by SimonCropp** [comment]
@andreasohlund i would say thats is a bug since if they have duplicate messages with the same OrderId  then every one of them will create a new saga

**Comment by andreasohlund** [comment]
I agree, lets raise an issue to track this? (since this is unlikely to
happen in the current version given that we never persist a saga with a
null/default corr id)

**Comment by SimonCropp** [comment]
@ramonsmits since u raised the initial question can u raise an issue in core.

@andreasohlund still some outstanding comments on the content of this PR

**Comment by andreasohlund** [comment]
@SimonCropp your comments have been addressed

**Pull request merged**